artemis-server/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/wireformat/ReplicationLargeMessageEndMessage.java

```diff
@@ -57,6 +64,7 @@ public void decodeRest(final ActiveMQBuffer buffer) {
       if (buffer.readableBytes() >= DataConstants.SIZE_LONG) {
          pendingRecordId = buffer.readLong();
       }
+      isDelete = buffer.readBoolean();
```

Comment:
can you check hasRemaining() on the buffer here, and return false by default?

We intend to keep live and replica the same version, but users do weird things... just in case please?